spark3/src/main/java/org/apache/iceberg/spark/source/SparkBatchScan.java

```diff
@@ -108,6 +111,12 @@ public Batch toBatch() {
     return this;
   }
 
+  @Override
+  public MicroBatchStream toMicroBatchStream(String checkpointLocation) {
```

Comment:
We may want to split `SparkBatchScan` that currently implements both `Scan` and `Batch` in the future (not now).

Reply:
yes. I guess that is the answer for my question here too: https://github.com/apache/iceberg/pull/2611#discussion_r635010805

Reply:
I think it is reasonable to try splitting in a follow-up PR. We can separate and reuse the `Scan` abstraction and then have independent hierarchies for `Batch` and `MicroBatch`.

Reply:
added this to my - **Next Steps** list in the PR description. Once, the PR concludes - I will create Issues for these.